Jellyfin.Plugin.SubtitleExtract/Providers/SubsMetadataProvider.cs

```diff
@@ -0,0 +1,109 @@
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly ILogger<SubsMetadataProvider> _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SubsMetadataProvider"/> class.
+        /// </summary>
+        /// <param name="subtitleEncoder">Instance of the <see cref="ISubtitleEncoder"/> interface.</param>
+        /// <param name="loggerFactory">Instance of the <see cref="ILoggerFactory"/> interface.</param>
+        /// <param name="logger">Instance of the <see cref="ILogger"/> interface.</param>
+        public SubsMetadataProvider(
+            ISubtitleEncoder subtitleEncoder,
+            ILoggerFactory loggerFactory,
+            ILogger<SubsMetadataProvider> logger)
+        {
+            _subtitleEncoder = subtitleEncoder;
+            _logger = logger;
+            _loggerFactory = loggerFactory;
+        }
+
+        /// <inheritdoc />
+        public string Name => "Jellyfin Subtitle Extractor";
+
+        /// <summary>
+        /// Gets the order in which the provider should be called. (Core provider is = 100).
+        /// </summary>
+        public int Order => 1000;
+
+        /// <inheritdoc/>
+        public bool HasChanged(BaseItem item, IDirectoryService directoryService)
+        {
+            if (item.IsFileProtocol)
+            {
+                var file = directoryService.GetFile(item.Path);
+                if (file != null && item.DateModified != file.LastWriteTimeUtc)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        /// <inheritdoc/>
+        public Task<ItemUpdateType> FetchAsync(Episode item, MetadataRefreshOptions options, CancellationToken cancellationToken)
+        {
+            return FetchSubtitles(item, cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public Task<ItemUpdateType> FetchAsync(Movie item, MetadataRefreshOptions options, CancellationToken cancellationToken)
+        {
+            return FetchSubtitles(item, cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public Task<ItemUpdateType> FetchAsync(Video item, MetadataRefreshOptions options, CancellationToken cancellationToken)
+        {
+            return FetchSubtitles(item, cancellationToken);
+        }
+
+        private async Task<ItemUpdateType> FetchSubtitles(BaseItem item, CancellationToken cancellationToken)
+        {
+            var config = SubtitleExtractPlugin.Current!.Configuration;
+
+            if (config.ExtractionDuringLibraryScan)
+            {
+                var extractor = new SubtitlesExtractor(_loggerFactory.CreateLogger<SubtitlesExtractor>(), _subtitleEncoder);
+                _logger.LogInformation("Extracting subtitles for: {Video}", item);
+
+                if (config.WaitExtraction)
+                {
+                    _ = extractor.Run(item, cancellationToken).ConfigureAwait(false);
```

Comment:
This has the potential to start many extractions in parallel, I don't think we should add this capability.

Reply:
Might be better to add them to a queue and process that way? Other solution might be to await all extractions, that might make library scans quite long even if subtitles extraction is a relatively fast process.

Reply:
I would await all extractions and default removal during scan to false

Reply:
Then we will do it that way.